clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -292,7 +292,10 @@ private Cluster getClusterForCurrentTopics(Cluster cluster) {
             unauthorizedTopics.retainAll(this.topics.keySet());
 
             for (String topic : this.topics.keySet()) {
-                partitionInfos.addAll(cluster.partitionsForTopic(topic));
+                List<PartitionInfo> partitionInfoList = cluster.partitionsForTopic(topic);
+                if (partitionInfoList != null) {
```

Comment:
This was necessary as deleting a matching topic causes a NullPointerException when KStreams is running.